scipy/stats/stats.py

```diff
@@ -237,6 +237,28 @@ def _chk2_asarray(a, b, axis):
     return a, b, outaxis
 
 
+def _contains_nan(a, nan_policy='propagate'):
+    if nan_policy not in ('propagate', 'raise', 'omit'):
+        raise ValueError("nan_policy must be either 'propagate', 'raise', or "
+                         "'ignore'")
+    try:
+        # Calling np.sum to avoid creating a huge array into memory
+        # e.g. np.isnan(a).any()
+        contains_nan = np.isnan(np.sum(a))
```

Comment:
Noticing now that ``np.sum(a)`` will emit a warning for invalid values, which is probably undesired here. So might want to wrap this inside

    with np.errstate(invalid='ignore'):